torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -55,21 +55,52 @@ Message deserialize(MessageType type, std::istream& is) {
 
 } // namespace
 
+void ProcessGroupAgent::collectNames() {
+  const std::string& workerName = workerId_.name_;
+  const auto worldSize = pg_->getSize();
+  const int64_t length = workerName.length();
+
+  // use c10d allgather to collect names
+  torch::Tensor nameTensor =
+      torch::zeros({WorkerId::MAX_NAME_LEN}, torch::kChar);
+  memcpy(nameTensor.storage().data(), workerName.c_str(), length);
+  std::vector<torch::Tensor> inputName = {nameTensor};
+  std::vector<std::vector<torch::Tensor>> outputNames(1);
+  for (int i = 0; i < worldSize; ++i) {
```

Comment:
nit: size_t

Reply:
why size_t here? `pg_->getSize()` returns an int.

Reply:
Oops didn't realize it returns `size`, usually the convention in C++ is to use `size_t` for sizes.